tests/expectations/test_expectation.py

```diff
@@ -251,3 +267,107 @@ def test_unrecognized_expectation_arg_raises_error():
         gxe.ExpectColumnMaxToBeBetween(
             column="foo", min_value=0, max_value=10, mostyl=0.95  # 'mostly' typo
         )
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "query",
+    [
+        pytest.param("SELECT * FROM table", id="no batch"),
+        pytest.param("SELECT * FROM {{ batch }}", id="invalid batch syntax 1"),
+        pytest.param("SELECT * FROM {batch}", id="invalid batch syntax 2"),
+    ],
+)
+def test_unexpected_rows_expectation_invalid_query_raises_error(query: str):
+    with pytest.raises(pydantic.ValidationError):
+        UnexpectedRowsExpectation(unexpected_rows_query=query)
+
+
+@pytest.fixture
+def taxi_db_path() -> str:
+    return file_relative_path(__file__, "../test_sets/quickstart/yellow_tripdata.db")
+
+
+@pytest.fixture
+def sqlite_datasource(
+    in_memory_runtime_context: AbstractDataContext, taxi_db_path: str
+) -> SqliteDatasource:
+    context = in_memory_runtime_context
+    datasource_name = "my_sqlite_datasource"
+    return context.sources.add_sqlite(
+        datasource_name, connection_string=f"sqlite:///{taxi_db_path}"
+    )
+
+
+@pytest.fixture
+def sqlite_batch(sqlite_datasource: SqliteDatasource) -> Batch:
+    datasource = sqlite_datasource
+    asset = datasource.add_table_asset("yellow_tripdata_sample_2022_01")
+
+    batch_request = asset.build_batch_request()
+    return asset.get_batch_list_from_batch_request(batch_request)[0]
+
+
+@pytest.mark.sqlite
+@pytest.mark.parametrize(
+    "query, expected_success, expected_observed_value, expected_unexpected_rows",
+    [
+        pytest.param(
+            "SELECT * FROM {active_batch} WHERE passenger_count > 7",
+            True,
+            0,
+            [],
+            id="success",
+        ),
+        pytest.param(
+            # There is a single instance where passenger_count == 7
+            "SELECT * FROM {active_batch} WHERE passenger_count > 6",
```

Comment:
Didn't you mention we're using {batch} instead of {active_batch}?

Reply:
yup that impacts more than this expectation so I'll have to follow up with another PR to change things across the board